In [1]:
%matplotlib inline
import pandas as pd

In [2]:
from IPython.core.display import HTML
css = open('style-table.css').read() + open('style-notebook.css').read()
HTML('<style>{}</style>'.format(css))

FileNotFoundError: [Errno 2] No such file or directory: 'style-table.css'

In [4]:
titles = pd.read_csv(r'C:\Users\HP\Desktop\LetsUpgrade\Day 16 , 17\titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

### How many movies are listed in the titles dataframe?

In [6]:
a

title  year
193222  Miss Jerry  1894

In [15]:
titles.shape[0]

244914

244914 movies are listed in the titles dataframes.

### What are the earliest two films listed in the titles dataframe?

In [32]:
#titles.sort_values(by=['year'], inplace=True)
titles.sort_values('year').tail(2)


title  year
183018  The Zero Century: Maetel  2026
242196                 100 Years  2115

### How many movies have the title "Hamlet"?

In [14]:
titles.title.value_counts()['Hamlet']


20

There are 20 movies having the title "Hamlet"

### How many movies are titled "North by Northwest"?

In [16]:
titles.title.value_counts()['North by Northwest']


1

There are 1 movies having the title "North by Northwest"

### When was the first movie titled "Hamlet" made?

In [ ]:
for i in titles.itertuples():
    #if i[0] == "Hamlet":
    
    print(i)
    for j in i:
        print(j)
   
    

In [23]:
for i in titles: 
  
    # printing the third element of the column 
    print (titles[i][2]) 
    print(i)

title
year


### List all of the "Treasure Island" movies from earliest to most recent.

In [3]:
titles.title.value_counts()['Treasure Island']


8

### How many movies were made in the year 1950?

In [30]:
temp = 0
for i in titles.year:
    if i == 1950:
        temp = temp + 1
print(temp)

1113


1113 movies were made in the year 1950

### How many movies were made in the year 1960?

In [31]:
temp = 0
for i in titles.year:
    if i == 1960:
        temp = temp + 1
print(temp)

1549


1549 movies were made in the year 1950

### How many movies were made from 1950 through 1959?

In [32]:
temp = 0
for i in titles.year:
    if i>=1950 and i<1959:
        temp = temp + 1
print(temp)

11419


11419 movies were made from 1950 through 1959

### In what years has a movie titled "Batman" been released?

### How many roles were there in the movie "Inception"?

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

### But how many roles in the movie "Inception" did receive an "n" value?

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

### How many roles were credited in the silent 1921 version of Hamlet?

### How many roles were credited in Branagh’s 1996 Hamlet?

### How many "Hamlet" roles have been listed in all film credits through history?

### How many people have played an "Ophelia"?

### How many people have played a role called "The Dude"?

### How many people have played a role called "The Stranger"?

### How many roles has Sidney Poitier played throughout his career?

### How many roles has Judi Dench played?

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

### List the leading roles that Cary Grant played in the 1940s in order by year.

### How many roles were available for actors in the 1950s?

### How many roles were available for actresses in the 1950s?

### How many leading roles (n=1) were available from the beginning of film history through 1980?

### How many non-leading roles were available through from the beginning of film history through 1980?

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?